### PR TITLE
Set up Raven + configs, add seed scripts

### PR DESCRIPTION
**Pull request by cywang117**
### Features
- Custom data generation script for RavenDB CSVs that include: `@id`, `@metadata.@collection` fields. Ids must be generated with a collection prefix, and as a string.
- (!doesn't work) Seeding script for RavenDB through client API. Crashes at 87% completion with heap memory errors. Ended up using the server GUI to upload my 3 CSV seeds manually. Possible fixes (which I'll implement if I end up choosing RavenDB, HIGHLY unlikely):
    - Start node process with increased --max-old-space-size flag
    - Split seeding into two sequential processes
    - Pause manually between bulkInsert calls to allow JS garbage collection to free up memory (doesn't make sense - `bulkInsert.finish` calls at end of each for loop should flush data, according to RavenDB's documentation)
    - Uninstall RavenDB

### Changes
- Remove CouchDB references in several files, replace with RavenDB
- Add some package.json scripts

### Review:
- `generateDescriptionsForRaven`, `generateGamesGenresForRaven` functions in `generateDescriptions.js` and `generateGamesGenres.js` - not much to see actually. They're the same as the original generate functions, just tweaked to generate workable CSVs for RavenDB.
- You _could_ review the RavenDB seed script (`seedRavenDB.js`) for craps & giggles if you want to. Since my hunch is that RavenDB's bulkInsert contains the memory leak I mentioned above, I'd be _very_ happy to hear about any memory leaks in my code. (No sarcasm - I genuinely want to know if I missed anything!)

[Journal detailing my struggles](https://www.notion.so/PR4-Seed-RavenDB-TODO-Write-queries-tests-1224600d2e144259b859cea566542436)
[Ticket](https://trello.com/c/oYhFReUI/5-l-learn-ravendb-seed-with-data-from-data-gen-script)

Many thanks! 🍌

### CODE REVIEW

File: db/data-gen/generateGamesGenres.js [hunk 1]
```diff
@@ -82,3 +83,95 @@ exports.generateGamesGenres = async (numRecords, numGenres, outDir) => {
...
+
+    // For each record, generate 3-5 genre pairs and convert to CSV before storing
+    let csvLinesForRecord = [];
+    let genreCount = getRandomInRange(3, 5);
+    usedGenres.length = 0;
+
+    for (let k = 0; k < genreCount; k++) {
+      let genreId;
+
+      do {
```
Comment (krisdo) [inline db/data-gen/generateGamesGenres.js:150]: I like your use of do-while loops!

File: db/seedRavenDB.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    path.resolve(__dirname, 'data-gen', 'csv');
+
+  let args = JSON.parse(process.env.npm_config_argv).original;
+  let csvDir = args.find(elt => /--dir=/.test(elt));
+
+  // Check if valid csvDir (ensuring path originates from project root instead of db/ dir)
+  if (csvDir) {
+    csvDir = path.resolve(__dirname, '..', csvDir.replace('--dir=', ''));
+    try {
+      await fs.promises.stat(csvDir);
```
Comment (krisdo) [inline db/seedRavenDB.js:167]: Christina, I am curious if it it's more a style to implementation when doing fs.promises.stat vs const fs = require('fs').promises in the beginning and using fs.stat later? I haven't seen many examples with the latter as I know the native api fs promises module do not have all of the fs methods.

**Review by krisdo — APPROVED** [review]
Sorry, that you had to spend so much time on seeding RavenDB. However, the GUI makes up for it! (I just browsed through their website)

Your code is clean and easy to read like usual. 👍

File: db/seedRavenDB.js (hunk 2 above)
Comment (cywang117) [reply db/seedRavenDB.js:167]: The first case would be if you had to use non-promise modules as well, but I don't think I use those in this file. It's a matter of preference 👍

**Pull request merged**